src/space_usage/mod.rs

```diff
@@ -0,0 +1,283 @@
+/*!
+Representations for the space usage of various parts of a Tantivy index.
+
+This can be used programmatically, and will also be exposed in a human readable fashion in
+tantivy-cli.
+
+One important caveat for all of this functionality is that none of it currently takes storage-level
+details into consideration. For example, if your file system block size is 4096 bytes, we can
+under-count actual resultant space usage by up to 4095 bytes per file.
+*/
+
+use schema::Field;
+use std::collections::HashMap;
+use std::ops::{Add, AddAssign};
+use serde::Serialize;
+use serde::Serializer;
+use serde::Deserialize;
+use serde::Deserializer;
+
+/// Indicates space usage in bytes
+#[derive(Clone, Copy, Debug)]
+pub struct ByteCount(pub usize);
```

Comment:
Do we want the inner value to be `pub` here?

Reply:
It might be worth investigating if there might be some usage of this struct elsewhere or if this is overkill.

Reply:
Yeah this should probably just be a `type ByteCount` instead of a `struct`.